dali/operators/generic/pad.h

```diff
@@ -85,12 +86,13 @@ class Pad : public Operator<Backend> {
 
     if (spec.ArgumentDefined("shape")) {
       GetShapeArgument(shape_, spec, "shape", ws, curr_batch_size);
+      shape_arg_provided = true;
     }
     if (spec.ArgumentDefined("align")) {
       GetShapeArgument(align_, spec, "align", ws, curr_batch_size);
     }
 
-    if (shape_.empty())
+    if (!shape_arg_provided)
```

Comment:
Can you write more in the PR description on how this fixes the bug?
Can the shape be provided and still be empty?

Reply:
I think it would be taken care of in line 105 - if `axes` are empty, they will be implicitly extended to cover all axes in the input (which, by the way, is not ideal) and the comparison will fail.

Reply:
I updated the description (__What happened in this PR__ section)